* drop hardcoded -O2 flag from CFLAGS
* drop hardcoded -s (strip) flag from LDFLAGS, and respect user LDFLAGS
* fix hardcoded `make` and hardcoded -j2 flags
* respect CFLAGS when linking

--- a/Makefile
+++ b/Makefile
@@ -13,9 +13,8 @@
 CC ?= gcc
 STRIP ?= strip
 
-CFLAGS := -O2 -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs $(CFLAGS)
+CFLAGS += -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs
 
-LDFLAGS = -s
 #LDFLAGS = -s -static
 INSTALL = install
 INSTALL_DATA = $(INSTALL) -m 644
@@ -24,11 +23,10 @@ INSTALL_PROGRAM = $(INSTALL)
 
 OBJS = hdparm.o identify.o sgio.o sysfs.o geom.o fallocate.o fibmap.o fwdownload.o dvdspeed.o wdidle3.o
 
-all:
-	make -j2 hdparm
+all: hdparm
 
 hdparm: hdparm.h sgio.h $(OBJS)
-	$(CC) $(LDFLAGS) -o hdparm $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o hdparm $(OBJS)
 	$(STRIP) hdparm
 
 hdparm.o:	hdparm.h sgio.h
